src/Compilers/CSharp/Portable/Compilation/CSharpCompilation.cs

```diff
@@ -3322,12 +3334,12 @@ public override bool ContainsSymbolsWithName(string name, SymbolFilter filter =
         /// CancellationToken)"/> when predicate is just a simple string check.  <paramref
         /// name="name"/> is case sensitive.
         /// </summary>
-        public override IEnumerable<ISymbol> GetSymbolsWithName(string name, SymbolFilter filter = SymbolFilter.TypeAndMember, CancellationToken cancellationToken = default(CancellationToken))
+        public override IEnumerable<ISymbol> GetSymbolsWithName(string name, SymbolFilter filter = SymbolFilter.TypeAndMember, CancellationToken cancellationToken = default)
         {
-            return GetSymbolsWithNameCore(name, filter, cancellationToken).GetPublicSymbols();
+            return GetSymbolsWithNameCore(name, filter, cancellationToken).GetPublicSymbols()!;
```

Comment:
Same question

Reply:
Same thing. We don't have a `NotNullWhenNotNull` for enumerables.